tests/unit/module_utils/core/test_core.py

```diff
@@ -0,0 +1,32 @@
+from plugins.module_utils.core import scrub_none_parameters
+import pytest
+
+scrub_none_test_data = [
+    (dict(),
+     dict()
+     ),
+    (dict(param1='something'),
+     dict(param1='something')
+     ),
+    (dict(param1='something', param2='something_else'),
+     dict(param1='something', param2='something_else')
+     ),
+    (dict(param1='something', param2=None),
+     dict(param1='something')
+     ),
+    (dict(param1='something', param2=None, param3=None),
+     dict(param1='something')
+     ),
+    (dict(param1='something', param2=None, param3=None, param4='something_else'),
+     dict(param1='something', param4='something_else')
+     ),
+    (dict(param1=None, param2=None),
+     dict()
+     )
+]
+
+
+@pytest.mark.parametrize("input_params, output_params", scrub_none_test_data)
```

Comment:
You don't have any tests for recursion, and especially not the condition that empty sub-dicts are also removed.

Reply:
I've added some additional tests to cover empty sub-dicts, sub-dicts as None and empty lists